src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -201,6 +201,14 @@ protected override bool TryParseRequest(ReadResult result, out bool endConnectio
         // Suppress pseudo headers from the public headers collection.
         HttpRequestHeaders.ClearPseudoRequestHeaders();
 
+        // Cookies should be merged into a single string separated by "; "
+        // https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.5
+        var headers = (AspNetCore.Http.IHeaderDictionary)HttpRequestHeaders;
```

Comment:
This cast isn't ideal. I looked at code-gen for the headers collection and I think that adding HeaderNames.Cookie to KnownHeaders.InternalHeaderAccessors will generate a `HeaderCookies` property to access the header with casting.

@Tratcher